src/psfmachine/machine.py

```diff
@@ -837,289 +635,197 @@ def build_time_model(self, plot=False, downsample=False, split_time_model=False)
-                flux_err_binned,
-                poscorr1_smooth,
-                poscorr2_smooth,
-                poscorr1_binned,
-                poscorr2_binned,
-            ) = self._time_bin(
-                npoints=self.n_time_points, downsample=self.downsample_time_knots
-            )
-        else:
-            (
-                time_original,
-                time_binned,
-                flux_binned_raw,
-                flux_binned,
-                flux_err_binned,
-            ) = self._time_bin(
-                npoints=self.n_time_points, downsample=self.downsample_time_knots
-            )
-
-        # not necessary to take value from Quantity to do .multiply()
-        dx, dy = (
-            self.uncontaminated_source_mask.multiply(self.dra),
-            self.uncontaminated_source_mask.multiply(self.ddec),
-        )
-        dx = dx.data * u.deg.to(u.arcsecond)
-        dy = dy.data * u.deg.to(u.arcsecond)
-
-        A_c = _make_A_cartesian(
-            dx,
-            dy,
-            n_knots=self.n_time_knots,
-            radius=self.time_radius,
-            spacing=self.cartesian_knot_spacing,
-        )
-        # if self.seg_splits.shape[0] == 2 and segment == 0:
-        #     seg_mask = np.ones(time_binned.shape[0], dtype=bool)
-        # else:
-        seg_mask = (time_binned[:, 0] >= time_original[self.seg_splits[segment]]) & (
-            time_binned[:, 0] < time_original[self.seg_splits[segment + 1] - 1]
-        )
-        # find the right mask that select the binned times andd flux
-        A2 = sparse.vstack([A_c] * time_binned[seg_mask].shape[0], format="csr")
-
-        if hasattr(self, "pos_corr1") and self.time_corrector in [
-            "pos_corr",
-            "centroid",
-        ]:
-            # Cartesian spline with poscor dependence
-            A3 = _combine_A(
-                A2, poscorr=[poscorr1_binned[seg_mask], poscorr2_binned[seg_mask]]
-            )
-        else:
-            # Cartesian spline with time dependence
-            A3 = _combine_A(A2, time=time_binned[seg_mask])
-
-        model = (
-            A3.dot(self.time_model_w[segment]).reshape(flux_binned[seg_mask].shape) + 1
-        )
-        fig, ax = plt.subplots(2, 2, figsize=(7, 6), facecolor="w")
-        k1 = (
-            self._time_masked[segment]
-            .reshape(flux_binned[seg_mask].shape)[0]
-            .astype(bool)
-        )
-        k2 = (
-            self._time_masked[segment]
-            .reshape(flux_binned[seg_mask].shape)[-1]
-            .astype(bool)
+        model_binned = self.P3.matrix.dot(self.P3.weights).reshape(
+            self.flux_binned.shape
         )
+        fig, ax = plt.subplots(2, 2, figsize=(9, 7), facecolor="w")
+        # k1 = self._time_masked_pix.reshape(self.flux_binned.shape)[0].astype(bool)
         im = ax[0, 0].scatter(
-            dx[k1],
-            dy[k1],
-            c=flux_binned[seg_mask][0][k1],
+            self.P3.dx[self._time_masked_pix],
```

Comment:
Let's use `self.dx`

Those should be the same, right?

Reply:
we don't have that attribute in `machine`